crates/engine/tree/src/persistence.rs

```diff
@@ -0,0 +1,80 @@
+use crate::tree::ExecutedBlock;
+use futures::ready;
+use reth_primitives::B256;
+use reth_provider::ProviderFactory;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::sync::mpsc::Receiver;
+
+// TODO: design questions:
+// * to support "unwinds" or not? ie, what do we do when we get a request to persist a chain that is
+// overlapping with the one on disk
+/// Writes parts of reth's in memory tree state to the database.
+pub struct Persistence<DB> {
+    /// The db / static file provider to use
+    provider: ProviderFactory<DB>,
+    // TODO: handles for pushing requests
+    /// Incoming requests to persist stuff
+    incoming: Receiver<PersistenceAction>,
+}
+
+impl<Writer> Persistence<Writer> {
+    // TODO: initialization
+    /// Writes the cloned tree state to the database
+    fn write(&mut self, blocks: Vec<ExecutedBlock>) {
+        todo!("implement this")
+    }
+}
+
+impl<Writer> Persistence<Writer>
+where
+    Writer: Unpin,
+{
```

Comment:
we expect that this will be spawned, right?

and internally this spawns regular std::thread to execute the actions?